metadata_layer.py

```diff
@@ -861,10 +868,10 @@ def mirror_link(self, model_object, data_source, link, link_obj, policy):
         # Mirror it.
         mirror.mirror_one(representation, mirror_url)
 
-        # If we couldn't mirror an open access link representation, suppress
+        # If we couldn't mirror an open/protected access link representation, suppress
         # the license pool until someone fixes it manually.
         if representation.mirror_exception:
-            if pools and link.rel == Hyperlink.OPEN_ACCESS_DOWNLOAD:
+            if pools and link.rel in [Hyperlink.OPEN_ACCESS_DOWNLOAD, Hyperlink.GENERIC_OPDS_ACQUISITION]:
```

Comment:
This list is used in multiple places, so might be useful to put it in a variable for DRY/maintainability.

Reply:
I agree, and  core/model/constants.py#LinkRelations is the appropriate place for this.

First, GENERIC_OPDS_ACQUISITION should be added to the MIRRORED list, which contains the link relations that a MirrorUploader might be expected to handle.

By definition, the list you're working with here is the intersection of  MIRRORED (should be run through a MirrorUploader) and CIRCULATION_ALLOWED (a book or book-like thing that can be circulated). So you can calculate it as a new list called something like MIRRORED_BOOKS, or you can just check both lists.

Reply:
I see that you already modified MIRRORED -- which makes sense since otherwise you'd get some test failures -- so the first part of my comment is moot.

Reply:
I added `SELF_HOSTED_BOOKS` to `Hyperlink` class and started using it